gprofiler/profilers/java.py

```diff
@@ -392,15 +492,25 @@ def _run_java_version() -> None:
 
     def _profile_process(self, process: Process) -> Optional[StackToSampleCount]:
         logger.info(f"Profiling process {process.pid} with async-profiler")
+        process_basename = os.path.basename(process.exe())
+        if self._java_safemode and process_basename != "java":
+            logger.error(f"Non-java basenamed process {process.pid} ({process.exe()}), skipping...")
+            return None
 
         # Get Java version
+        java_version_output = self._get_java_version(process)
+
         # TODO we can get the "java" binary by extracting the java home from the libjvm path,
         # then check with that instead (if exe isn't java)
-        if self._version_check and os.path.basename(process.exe()) == "java":
-            if not self._is_jdk_version_supported(self._get_java_version(process)):
-                logger.warning(f"Process {process.pid} running unsupported Java version, skipping...")
+        if self._version_check and process_basename == "java":
+            if not self._is_jvm_type_supported(java_version_output):
+                logger.error(f"Process {process.pid} running unsupported JVM, skipping...")
                 return None
 
+        if self._java_safemode and not self._is_jvm_version_supported(java_version_output):
+            logger.error(f"Process {process.pid} running unsupported Java version, skipping...")
```

Comment:
```suggestion
            logger.error(f"Process {process.pid} running unsupported Java version, skipping... (disable --java-safemode to profile it anyway")
```
or something like that